docs/enhancements/infrastructure-monitoring-service-api/lifecycle.md

```diff
@@ -0,0 +1,530 @@
+| `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions`                       | POST            | Create a new alarm subscriptions.                                   | `AlarmSubscriptionInfo`                                                      | Exactly one `AlarmSubscriptionInfo` |
+| `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions/{alarmSubscriptionId}` | GET             | Retrieve exactly one subscription using `alarmSubscriptionId`.      | None                                                                         | Exactly one `AlarmSubscriptionInfo` |
+| `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions/{alarmSubscriptionId}` | DELETE          | Delete exactly one subscription using `alarmSubscriptionId`.        | None                                                                         | None                                |
+
+
+
+| **Internal Endpoint**                           | **HTTP Method** | **Description**                               | **Input Payload**                                                        | **Returned Data** |
+|-------------------------------------------------|-----------------|-----------------------------------------------|--------------------------------------------------------------------------|-------------------|
+| `internal/v1/caas-alerts/alertmanager`          | POST            | Alertmanager notifications comes through here | https://prometheus.io/docs/alerting/latest/configuration/#webhook_config | None              |
+| `internal/v1/hardware-alerts/{hw-vendor-name}}` | POST            | TBD                                           | TBD                                                                      | TBD               |
+
+
+
+### `alarms` family
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarms` with GET
+1. Get active alarms from `alarm_event_record` table
+3. Response with retrieved list of AlarmEventRecord
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarms/{alarmEventRecordId}` with GET
+1. Client calls with an AlarmEventRecordID
+2. Get active alarms from DB from `alarm_event_record`
+3. Response with retrieved instance of AlarmEventRecord
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarms/{alarmEventRecordId}` with PATCH
+1. Client calls with an AlarmEventRecordID and `AlarmEventRecordModifications` as patch payload
+2. If AlarmEventRecordModifications.alarmAcknowledged is True, update `alarm_event_record` table
+3. Response with `AlarmEventRecordModifications`
+
+### `alarmSubscriptions` family
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions` with GET
+1. Query the storage `alarm_subscription_info`
```

Comment:
also need filter the records if filter is present.

Reply:
updated doc to include filter